### PR TITLE
Make woodpecker run steps in parallel

### PR DESCRIPTION
**Pull request by SleeplessOne1917**

### CODE REVIEW

File: .woodpecker.yml [hunk 1]
```diff
@@ -1,56 +1,35 @@
...
 
 steps:
-  prepare_repo:
-    image: alpine:3
-    commands:
-      - apk add git
-      - git submodule init
-      - git submodule update
-    when:
-      - event: [pull_request, tag]
```
Comment (SleeplessOne1917) [inline .woodpecker.yml:12]: Removed because this repo doesn't have submodules.

File: .woodpecker.yml [hunk 2]
```diff
@@ -1,56 +1,35 @@
...
-
-  prettier_check:
-    image: tmknom/prettier:3.2.5
-    commands:
-      - prettier -c .
-    when:
-      - event: pull_request
```
Comment (SleeplessOne1917) [inline .woodpecker.yml:19]: Removed because none of the file types in this project are affected by prettier.

File: .woodpecker.yml [hunk 3]
```diff
@@ -1,56 +1,35 @@
 variables:
-  - &rust_image "rust:1.78"
+  - &rust_image "rust:1.78-slim-bookworm"
```
Comment (dessalines) [inline .woodpecker.yml:2]: Might we worth a try to see if alpine images work here too.

File: .woodpecker.yml [hunk 4]
```diff
@@ -1,56 +1,35 @@
...
-    when:
-      - event: pull_request
-
   toml_fmt:
     image: tamasfe/taplo:0.8.1
     commands:
       - taplo format --check
     when:
       - event: pull_request
+    depends_on: [] # enable parallel steps
```
Comment (dessalines) [inline .woodpecker.yml:11]: > By setting depends_on on a single step all other steps will be immediately executed as well if no further dependencies are specified.

I spose this should be enough then.

**Pull request merged**